microovn/ovn/environment.go

```diff
@@ -179,3 +182,55 @@ func createPaths() error {
 
 	return nil
 }
+
+// cleanupPaths backs up directories defined by paths.BackupDirs and then removes directories
+// created by createPaths function. This effectively removes any data created during MicroOVN runtime.
+func cleanupPaths() []error {
+	var errors []error
+
+	// Create timestamped backup dir
+	backupDir := fmt.Sprintf("backup_%d", time.Now().Unix())
+	backupPath := filepath.Join(paths.Root(), backupDir)
+	err := os.Mkdir(backupPath, 0750)
+	if err != nil {
+		errors = append(
+			errors,
+			fmt.Errorf(
+				"failed to create backup directory '%s'. Refusing to continue with data removal: %s",
+				backupPath,
+				err,
+			),
+		)
+		return errors
+	}
+
+	// Backup selected directories
+	for _, dir := range paths.BackupDirs() {
+		_, fileName := filepath.Split(dir)
+		destination := filepath.Join(backupPath, fileName)
+		err = os.Rename(dir, destination)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	// Return if any backups failed
+	if len(errors) > 0 {
+		errors = append(
+			errors,
+			fmt.Errorf("failures occured furing backup. Refusing to continue with data removal"),
```

Comment:
Following on the comment at the begining of the function, I'd move this message to where `cleanupPaths` is called after bailing from this function on the first error.

Reply:
But errors can happen for other reasons than failed backup. If we logged this message outside of a context of this function, we'd have to somehow pass error "reason" to the caller.